src/dataIntegration/transcriptomics/infos_and_TODO.md

```diff
@@ -0,0 +1,53 @@
+
+call_fastcore.m
+  -	setting of the default values for the algorithm parameters : tol, core, scaling
+  -	the initial algorithm has been developped for CPLEX and we have adapated the code for gurobi? need to check the compatibility with other solver and propose options for users to manage that
+
+call_GIMME.m
+  -	setting of the default values for the algorithm parameters : tol, obj_fraction
+  -	provide an option to modulate the tolerance used in findBlockedReaction (check the last version of this code in cobra v3) - findBlocked reaction is used to check model consistency after removal of reactions identified by GIMME
+  -	a modified version of convertToIrreversible is introduced in this script, need to check the fixes introduced to the last version that will be available in cobra v3 to see if we can remove that
+
+call_iMAT.m
+  -	setting of the default values for the algorithm parameters 
+  -	provide an option to modulate the tolerance used in findBlockedReaction (check the last version of this code in cobra v3) - findBlocked reaction is used to check model consistency after removal of reactions identified by iMAT
+  -	the algorithm use a modified version of solveCobraMILP when the computation time limit is reached ? can be implemented in the general file of solveconraMILP
+
+call_INIT.m
+  -	setting of the default values for the algorithm parameters 
+  -	provide an option to modulate the tolerance used in findBlockedReaction (check the last version of this code in cobra v3) - findBlocked reaction is used to check model consistency after removal of reactions identified by INIT
+  -	the algorithm use a modified version of solveCobraMILP when the computation time limit is reached ? can be implemented in the general file of solveconraMILP
+  -	need to add line to compute the ?weights?  using the expression data if the user do no want to provide the weights manually
+
+call_MBA.m
+  -	setting of the default values for the algorithm parameters 
+  -	update use of CheckModelConsistency.m see the second comment of  "General" section below
+
+call_mCADRE.m
+  -	setting of the default values for the algorithm parameters 
+  -	update use of CheckModelConsistency.m see the second comment of  "General" section below
+  -	include a local copy of changeRxnBounds such as to not print warning if a reaction has been removed? could be suppressed by introducing "warning off" before the call of this function
+  -	!!!! lot of writing improvements can be done on mCADRE code? still working on it
+
+
+General
+  -	most of the call_xx function ask for a threshold value for expression data - need to write an additional code that allow to identify different threshold value depending on the expression data available if they do not want to set it manually (e.g. 75 percentile of the average expression data)
```

Comment:
Actually I would suggest three different types of inputs:
A) -1 / 0 / 1 indicators for "off/unknown/on" 
B) Expression values with a numeric cutoff for "OFF/ON"
C) Expression values and a percentage cutoff for "OFF/ON".
For b and C 2 different cutoffs could be supplied to allow potentially converting to -1/0/1 as above.

Now there will be some methods that actually use the expression value (e.g. GIMME), while others only use nominal indicators (like MBA), and the function should check whether the input arguments fit. e.g. for MBA, lower and upper cutoffs are necessary to indicate the "medium" set. For GIMME, only an upper cutoff is needed.

I would also suggest to keep the "discretized" GIMME version currently in the toolbox but to allow selection of either discretized or continuous gimme. 

The interesting questions would be: what are the default cutoffs. 
Personally I would suggest the current > 75 percentile  -> ON ; <25 percentile -> OFF. But thats to discuss.